src/Bicep.Core/Analyzers/Linter/LinterRuleBase.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    public abstract class LinterRuleBase : IBicepAnalyzerRule
+    {
+        public LinterRuleBase(string code, string description, string docUri,
+                          DiagnosticLevel diagnosticLevel = DiagnosticLevel.Warning,
+                          DiagnosticLabel? diagnosticLabel = null)
+        {
+            this.AnalyzerName = LinterAnalyzer.AnalyzerName;
+            this.Code = code;
+            this.Description = description;
+            this.DocumentationUri = docUri;
+            this.DiagnosticLevel = diagnosticLevel;
+            this.DiagnosticLabel = diagnosticLabel;
+        }
+
+        internal const string FailedRuleCode = "Linter Rule Error";
+        private IConfigurationRoot? Config;
+        public string AnalyzerName { get; }
+
+        public string Code { get; }
+        public readonly string RuleConfigSection = $"{LinterAnalyzer.SettingsRoot}:{LinterAnalyzer.AnalyzerName}:rules";
+        public bool Enabled => this.DiagnosticLevel != DiagnosticLevel.Off;
```

Comment:
>Enabled [](http://example.com/codeflow?start=20&length=7)

It's really weird that we would allow a rule implementation to decide what enabled means. Can we move this out of the rule contract?

Reply:
move to IsEnabled() extension method